docs/resources/appdata_dsource.md

```diff
@@ -0,0 +1,176 @@
+        * `password` - Password to assign to the environment variables.
+        * `vault` - The name or reference of the vault to assign to the environment variables.
+        * `hashicorp_vault_engine` - Vault engine name where the credential is stored.
+        * `hashicorp_vault_secret_path` -  Path in the vault engine where the credential is stored.
+        * `hashicorp_vault_username_key` - Hashicorp vault key for the username in the key-value store.
+        * `hashicorp_vault_secret_key` - Hashicorp vault key for the password in the key-value store.
+        * `azure_vault_name` - Azure key vault name.
+        * `azure_vault_username_key` - Azure vault key in the key-value store.
+        * `azure_vault_secret_key` - Azure vault key in the key-value store.
+        * `cyberark_vault_query_string` - Query to find a credential in the CyberArk vault.
+    
+* `ops_pre_sync` - (Optional) Operations to perform after syncing a created dSource.
+    * `name` - Name of the hook
+    * `command` - (Required)Command to be executed
+    * `shell` - Type of shell. Valid values are `[bash, shell, expect, ps, psd]` 
+    * `credentials_env_vars` - List of environment variables that will contain credentials for this operation
+        * `base_var_name` - Base name of the environment variables. Variables are named by appending '_USER', '_PASSWORD', '_PUBKEY' and '_PRIVKEY' to this base name, respectively. Variables whose values are not entered or are not present in the type of credential or vault selected, will not be set.
+        * `password` - Password to assign to the environment variables.
+        * `vault` - The name or reference of the vault to assign to the environment variables.
+        * `hashicorp_vault_engine` - Vault engine name where the credential is stored.
+        * `hashicorp_vault_secret_path` -  Path in the vault engine where the credential is stored.
+        * `hashicorp_vault_username_key` - Hashicorp vault key for the username in the key-value store.
+        * `hashicorp_vault_secret_key` - Hashicorp vault key for the password in the key-value store.
+        * `azure_vault_name` - Azure key vault name.
+        * `azure_vault_username_key` - Azure vault key in the key-value store.
+        * `azure_vault_secret_key` - Azure vault key in the key-value store.
+        * `cyberark_vault_query_string` - Query to find a credential in the CyberArk vault.
+
+* `excludes` - (Optional) List of subdirectories in the source to exclude when syncing data.These paths are relative to the root of the source directory. [AppDataDirect only]
+
+* `parameters` - (Optional) The JSON payload conforming to the DraftV4 schema based on the type of application data being manipulated.
```

Comment:
"The JSON payload is based on the type of dSource being created. Different parameters apply to different data sources."